src/appleseed-max-impl/builtinmapsupport.cpp

```diff
@@ -43,6 +44,47 @@
 namespace asf = foundation;
 namespace asr = renderer;
 
+void connect_output_selector(
+    renderer::ShaderGroup&  shader_group,
+    const char*             material_node_name,
+    const char*             material_input_name,
+    Texmap*                 texmap,
+    const Color             const_value)
+{
+    const auto t = GetCOREInterface()->GetTime();
+
+    Texmap* input_map = nullptr;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"source_map", t, input_map, FOREVER);
+    int output_index = 0;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"output_index", t, output_index, FOREVER);
+
+    if (input_map != nullptr && is_osl_texture(input_map))
+    {
+        auto output_names = static_cast<OSLTexture*>(input_map)->get_output_names();
+
+        static_cast<OSLTexture*>(input_map)->create_osl_texture(
+            shader_group,
+            material_node_name,
+            material_input_name,
+            static_cast<int>((output_index - 1) % output_names.size()));
```

Comment:
What is the purpose of the modulo operation here? It looks suspicious :)

Reply:
`output_index` should not be more than `output_names.size()` but instead of constraining the parameter I thought I'll use modulo so that any value can be used as an index for the names vector.
I've seen similar behavior somewhere else in Max. I'll try to specify in slack..

Reply:
Material IDs of the meshes work in the same way: let's say we have mesh with face id-s 1, 2, 3. I can apply multi-material with 4 sub-materials to that mesh. 4th sub-material will be applied to the face with ID = 1.
Even when some index parameter has value bigger than collection it indexes, instead of discarding bigger values they are still used.
Hope it's clear.